MvvmCross/Windows/Wpf/Views/Presenters/MvxWpfViewPresenter.cs

```diff
@@ -0,0 +1,202 @@
+                };
+            }
+            window.Closed += Window_Closed;
+            _frameworkElementsDictionary.Add(window, new Stack<FrameworkElement>());
+
+            if (!(element is Window))
+            {
+                _frameworkElementsDictionary[window].Push(element);
+                window.Content = element;
+            }
+
+            if (attribute.Modal)
+                window.ShowDialog();
+            else
+                window.Show();
+        }
+
+        private void Window_Closed(object sender, EventArgs e)
+        {
+            var window = sender as Window;
+            window.Closed -= Window_Closed;
+
+            if (_frameworkElementsDictionary.ContainsKey(window))
+                _frameworkElementsDictionary.Remove(window);
+        }
+
+        protected virtual void ShowContentView(FrameworkElement element, MvxContentPresentationAttribute attribute, MvxViewModelRequest request)
+        {
+            var window = _frameworkElementsDictionary.Keys.FirstOrDefault(w => (w as MvxWindow)?.Identifier == attribute.WindowIdentifier) ?? _frameworkElementsDictionary.Keys.Last();
+            _frameworkElementsDictionary[window].Push(element);
+            window.Content = element;
+        }
+
+        public override void Close(IMvxViewModel toClose)
+        {
+            // toClose is window
+            if (_frameworkElementsDictionary.Any(i => (i.Key as IMvxWpfView)?.ViewModel == toClose) && CloseWindow(toClose))
+                return;
+
+            // toClose is content
+            if (_frameworkElementsDictionary.Any(i => (i.Value.Peek() as IMvxWpfView)?.ViewModel == toClose) && CloseContent(toClose))
+                return;
+
+            MvxTrace.Warning($"Could not close ViewModel type {toClose.GetType().Name}");
+        }
+
+        protected virtual bool CloseWindow(IMvxViewModel toClose)
+        {
+            var item = _frameworkElementsDictionary.FirstOrDefault(i => (i.Key as IMvxWpfView)?.ViewModel == toClose);
+            var window = item.Key;
+            window.Close();
+            _frameworkElementsDictionary.Remove(window);
+
+            return true;
+        }
+
+        protected virtual bool CloseContent(IMvxViewModel toClose)
+        {
+            var item = _frameworkElementsDictionary.FirstOrDefault(i => (i.Value.Peek() as IMvxWpfView)?.ViewModel == toClose);
+            var window = item.Key;
+            var elements = item.Value;
+
+            if (elements.Any())
+                elements.Pop(); // Pop closing view
+
+            if (elements.Any())
+            {
+                window.Content = elements.Peek();
+                return true;
+            }
+
+            // Close window if no contents
+            _frameworkElementsDictionary.Remove(window);
+            window.Close();
+
+            return true;
+        }
+
+        protected MvxBasePresentationAttribute GetPresentationAttributes(FrameworkElement element)
```

Comment:
The other presenters use the ViewModel type to get the attribute. This might be different on WPF, just want to check: https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross/Droid/Droid/Views/MvxAndroidViewPresenter.cs#L189